File: backend/python-ai/src/models/chapter.py
"""章节数据模型"""

import uuid
from typing import TYPE_CHECKING

from sqlalchemy import ForeignKey, Integer, String, Text, UniqueConstraint
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from .base import BaseModel

if TYPE_CHECKING:
    from .story import Story


class Chapter(BaseModel):
    """章节"""

    __tablename__ = "chapters"
    __table_args__ = (
        UniqueConstraint("story_id", "chapter_number", name="uq_chapter_story_number"),
    )

    story_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("stories.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    chapter_number: Mapped[int] = mapped_column(Integer, nullable=False)
    title: Mapped[str | None] = mapped_column(String(200), nullable=True)
    content: Mapped[str | None] = mapped_column(Text, nullable=True)
    content_url: Mapped[str | None] = mapped_column(Text, nullable=True)
    word_count: Mapped[int | None] = mapped_column(Integer, nullable=True)
    status: Mapped[str] = mapped_column(
        String(20),
        nullable=False,
        default="draft",
        server_default="draft",
    )

    # Relationships
    story: Mapped["Story"] = relationship(back_populates="chapters")

    def __repr__(self) -> str:
        return f"<Chapter id={self.id} number={self.chapter_number} title={self.title!r}>"

File: backend/java-service/src/main/java/com/dreamweaver/entity/Chapter.java
package com.dreamweaver.entity;

import java.time.OffsetDateTime;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

@Entity
@Table(
    name = "chapters",
    uniqueConstraints = @UniqueConstraint(name = "uq_chapter_story_number", columnNames = {
        "story_id",
        "chapter_number"
    })
)
public class Chapter extends BaseEntity {

    @Column(name = "story_id", nullable = false)
    private UUID storyId;

    @Column(name = "chapter_number", nullable = false)
    private Integer chapterNumber;

    @Column(length = 200)
    private String title;

    @Column(columnDefinition = "text")
    private String content;

    @Column(name = "content_url", columnDefinition = "text")
    private String contentUrl;

    @Column(name = "word_count")
    private Integer wordCount;

    @Column(nullable = false, length = 20)
    private ChapterStatus status = ChapterStatus.DRAFT;

    @Column(name = "workflow_stage", nullable = false, length = 50)
    private ChapterWorkflowStage workflowStage = ChapterWorkflowStage.CHAPTER_CREATED;

    @Column(name = "confirmed_at")
    private OffsetDateTime confirmedAt;

    @Column(name = "last_generation_id")
    private UUID lastGenerationId;

    public UUID getStoryId() {
        return storyId;
    }

    public void setStoryId(UUID storyId) {
        this.storyId = storyId;
    }

    public Integer getChapterNumber() {
        return chapterNumber;
    }

    public void setChapterNumber(Integer chapterNumber) {
        this.chapterNumber = chapterNumber;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getContentUrl() {
        return contentUrl;
    }

    public void setContentUrl(String contentUrl) {
        this.contentUrl = contentUrl;
    }

    public Integer getWordCount() {
        return wordCount;
    }

    public void setWordCount(Integer wordCount) {
        this.wordCount = wordCount;
    }

    public ChapterStatus getStatus() {
        return status;
    }

    public void setStatus(ChapterStatus status) {
        this.status = status;
    }

    public ChapterWorkflowStage getWorkflowStage() {
        return workflowStage;
    }

    public void setWorkflowStage(ChapterWorkflowStage workflowStage) {
        this.workflowStage = workflowStage;
    }

    public OffsetDateTime getConfirmedAt() {
        return confirmedAt;
    }

    public void setConfirmedAt(OffsetDateTime confirmedAt) {
        this.confirmedAt = confirmedAt;
    }

    public UUID getLastGenerationId() {
        return lastGenerationId;
    }

    public void setLastGenerationId(UUID lastGenerationId) {
        this.lastGenerationId = lastGenerationId;
    }
}
